service/src/main/java/bio/terra/workspace/service/workspace/flight/create/workspace/WorkspaceCreateFlight.java

```diff
@@ -62,15 +51,6 @@ public WorkspaceCreateFlight(FlightMap inputParameters, Object applicationContex
               new CreateWorkspacePoliciesStep(
                   workspace, policyInputs, appContext.getTpsApiDispatch(), userRequest),
               serviceRetryRule);
-          if (cloningInstructions != CloningInstructions.COPY_NOTHING) {
-            addStep(
-                new MergePolicyAttributesStep(
-                    sourceWorkspaceUuid,
-                    workspace.workspaceId(),
-                    cloningInstructions,
-                    appContext.getTpsApiDispatch()),
-                serviceRetryRule);
-          }
```

Comment:
This is the workspace creation flight, it does not make sense to deal with cloning operations here. Leave that to the clone workspace flight.

@ddietterich let me know if there is a reason this was here. The clone flight also has it but I had to fix the cloning instructions (see above).

Reply:
The create workspace flight is used by clone workspace to create the target workspace.
Please do not delete that.

Reply:
Ah, I see. There is a problem, but it is that the `CloneWorkspaceFlight` is doing the merge. I think the one [here](https://github.com/DataBiosphere/terra-workspace-manager/blob/939ab7cd4c6374cdcfe212a942241ff6d8151266/service/src/main/java/bio/terra/workspace/service/resource/controlled/flight/clone/workspace/CloneWorkspaceFlight.java#L67) should be removed.

The step in `WorkspaceCreateFlight` is properly choosing to link or merge depending on the inputs. The one is `CloneWorkspaceFlight` is always merging.

Reply:
I agree that `CloneWorkspaceFlight` is doing the wrong thing and I fixed it in this PR. I don't think it is expected that `WorkspaceCreateFlight` has anything to do with it though. So I would opt for removing it from `WorkspaceCreateFlight` and doing it correctly in `CloneWorkspaceFlight` as proposed in this PR

Reply:
I'm not seeing the incorrectness. Both workspace clone and create run the WorkspaceCreateFlight. I guess we could rename it CreateWorkspaceForNewOrCloneFlight, to make it more clear that it is a flight that is shared by multiple cases.